src/entry_point/mod.rs

```diff
@@ -120,21 +120,20 @@ unsafe extern "C" fn rust_start(app_start: usize, stacktop: usize, _app_heap_bre
     // `sbrk` system call to dynamically request heap memory from the kernel, we
     // need to tell `linked_list_allocator` where the heap starts and ends.
     //
-    // Heap size is set using `elf2tab` with `--app-heap` option, which is
-    // currently at 1024. If you change the `elf2tab` heap size, make sure to
-    // make the corresponding change here.
-    const HEAP_SIZE: usize = 1024;
+    // We get this from the environment to make it easy to set per compile.
+    // You will get a compile error here if the variable isn't set!
+    let app_heap_size: usize = env!("APP_HEAP_SIZE").parse::<usize>().unwrap();
```

Comment:
Isn't this going to do the parsing at runtime? I think this should be done at compile time if at all possible.

Reply:
No, env! works at compile time https://doc.rust-lang.org/1.2.0/std/macro.env!.html

Reply:
It's the `.parse::<usize>().unwrap()` part I'm concerned about.

Reply:
oh bleh you are right. I'll see if I can move that to compile time. My best idea for how to do that is a procedural macro.